test/helpers/foundry/options.ts

```diff
@@ -0,0 +1,142 @@
+`);
+}
+
+export function parseArgs(args: string[] = argv.slice(2)) {
+  const { $0, _, ...parsed } = yargs()
+    // Ensure unrecognized commands/options are reported as errors.
+    .strict()
+    // disable yargs's version, as it doesn't make sense here
+    .version(false)
+    // use the scriptName in `--help` output
+    .scriptName('yarn foundryup')
+    // wrap output at a maximum of 120 characters or `stdout.columns`
+    .wrap(Math.min(120, stdout.columns))
+    .parserConfiguration({
+      'strip-aliased': true,
+      'strip-dashed': true,
+    })
+    // enable ENV parsing, which allows the user to specify foundryup options
+    // via environment variables prefixed with `FOUNDRYUP_`
+    .env('FOUNDRYUP')
+    .command(['$0', 'install'], 'Install foundry binaries', (builder) => {
+      builder.options(getOptions()).pkgConf('foundryup');
+    })
+    .command('cache', '', (builder) => {
+      builder.command('clean', 'Remove the shared cache files').demandCommand();
+    })
+    .parseSync(args);
+
+  const command = _.join(' ');
+  if (command === 'cache clean') {
+    return {
+      command,
+    } as const;
+  }
+
+  // if we get here `command` is always 'install' or '' (yargs checks it)
+  return {
+    command: 'install',
+    options: parsed as ParsedOptions<ReturnType<typeof getOptions>>,
+  } as const;
+}
+
+const Binaries = Object.values(Binary) as BinariesTuple;
+
+function getOptions(
+  defaultPlatform = platform(),
+  defaultArch = normalizeSystemArchitecture(),
+) {
+  return {
+    binaries: {
+      alias: 'b',
+      type: 'array' as const,
+      multiple: true,
+      description: 'Specify the binaries to install',
+      default: Binaries,
+      choices: Binaries,
+      coerce: (values: Binary[]): Binary[] => [...new Set(values)], // Remove duplicates
+    },
+    checksums: {
+      alias: 'c',
+      description: 'JSON object containing checksums for the binaries.',
+      coerce: (rawChecksums: string | Checksums): Checksums => {
+        try {
+          return typeof rawChecksums === 'string'
+            ? JSON.parse(rawChecksums)
+            : rawChecksums;
+        } catch {
+          throw new Error('Invalid checksums');
+        }
+      },
+      optional: true,
+    },
+    repo: {
+      alias: 'r',
+      description: 'Specify the repository',
+      default: 'foundry-rs/foundry',
+    },
+    version: {
+      alias: 'v',
+      description:
+        'Specify the version (see: https://github.com/foundry-rs/foundry/tags)',
+      default: 'nightly',
+      coerce: (
+        rawVersion: string,
+      ): { version: 'nightly' | `v${string}`; tag: string } => {
+        if (/^nightly/u.test(rawVersion)) {
+          return { version: 'nightly', tag: rawVersion };
+          // we don't validate the version much, we just trust the user
+        } else if (/^\d/u.test(rawVersion)) {
+          return { version: `v${rawVersion}`, tag: rawVersion };
+        }
```

Comment:
This chunk was ported from the bash script. It's weird though, because foundry doesn't publish tagged versions... only `nightly-<commit hash>` and `nightly` tags. Kept it here just to maintain feature parity.